### PR TITLE
Add docstring to Nexus

### PR DESCRIPTION
**Pull request by souravsingh**
Fixes a part of #1203

### CODE REVIEW

**Comment by peterjc** [comment]
The indentation is wrong, as can be seen from the TravisCI style checks:

```
flake8 Bio/
Bio/Nexus/Nexus.py:80:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:81:8: E999 IndentationError: unexpected indent
Bio/Nexus/Nexus.py:81:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:92:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:93:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:102:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:103:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:107:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:108:1: W293 blank line contains whitespace
Bio/Nexus/Nexus.py:110:10: W291 trailing whitespace
Bio/Nexus/Nexus.py:111:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:124:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:125:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:186:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:187:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:213:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:214:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:852:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:853:9: E113 unexpected indentation
Bio/Nexus/Nexus.py:937:7: E111 indentation is not a multiple of four
Bio/Nexus/Nexus.py:938:9: E113 unexpected indentation
```

This indirectly is causing the confusing mass of failures on TravisCI for the actual tests - the install seems to break part way through with ``Sorry: IndentationError: unexpected indent (Nexus.py, line 81)`` but TravisCI continues anyway.

File: Bio/Nexus/Nexus.py [hunk 1]
```diff
@@ -934,7 +934,7 @@ def _statelabels(self, options):
         pass
 
     def _matrix(self, options):
-        """ Creates a matrix for NEXUS object."""
+        """Creates a matrix for NEXUS object."""
```
Comment (peterjc) [inline Bio/Nexus/Nexus.py]: This is a private method (starts with an underscore, not one of the underscore-underscore special methods like ``__len__``), so could you add ``... (PRIVATE)."""``` to the end of the line:

```python
   """Creates a matrix for NEXUS object (PRIVATE)."""
```

File: Bio/Nexus/Nexus.py [hunk 2]
```diff
@@ -934,7 +934,7 @@ def _statelabels(self, options):
         pass
 
     def _matrix(self, options):
-        """Creates a matrix for NEXUS object."""
+        """Creates a matrix for NEXUS object (PRIVATE)"""
```
Comment (peterjc) [inline Bio/Nexus/Nexus.py]: And the trailing full stop too please.

File: Bio/Nexus/Nexus.py [hunk 3]
```diff
@@ -1028,6 +1043,7 @@ def _matrix(self, options):
             "Please Report this as a bug, and send in data file."
 
     def _translate(self, options):
+        """Translates a Nexus file (PRIVATE).""""
```
Comment (peterjc) [inline Bio/Nexus/Nexus.py]: There are four double-quotes at the end of that line, which is where this flake8 error comes from:

```
$ flake8 Bio/
Bio/Nexus/Nexus.py:1046:49: E999 SyntaxError: EOL while scanning string literal
...
```

**Comment by peterjc** [comment]
Bye eye this looked good, but:

```
$ flake8 Bio/
Bio/Nexus/Nexus.py:1046:49: E999 SyntaxError: EOL while scanning string literal
Bio/Nexus/Nexus.py:1109:67: W291 trailing whitespace
Bio/Nexus/Nexus.py:1122:54: W291 trailing whitespace
```

**Comment by souravsingh** [comment]
@peterjc Is the patch ready to merge?

**Comment by peterjc** [comment]
There are a few places where I'm not 100% sure what the method actually does and if your description is exactly correct, but your docstrings are a big improvement. Thanks!

**Pull request merged**